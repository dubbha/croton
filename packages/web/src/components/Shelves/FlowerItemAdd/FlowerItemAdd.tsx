import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import { Card } from 'react-bootstrap';

import '../styles.scss';

type Props = {
  onSelect: () => void
}

export const FlowerItemAdd = ({ onSelect }: Props) => {
  return (
    <div className="flower-item">
      <Card text="primary" onClick={onSelect}>
        <div className="shelf-item-new">
          <Card.Title>
            <FontAwesomeIcon icon={faPlus} />
            <span className="icon-prefix">Add New Flower</span>
          </Card.Title>
        </div>
      </Card>
    </div>
  );
};
