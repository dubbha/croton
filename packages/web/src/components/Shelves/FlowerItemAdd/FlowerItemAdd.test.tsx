import React from 'react';
import { render } from '@testing-library/react';
import { FlowerItemAdd } from '../FlowerItemAdd';

describe('components/FlowerItemAdd', () => {
  const props = {
    onSelect: () => {},
  };

  it('should render successfully', () => {
    const { container } = render(<FlowerItemAdd {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
