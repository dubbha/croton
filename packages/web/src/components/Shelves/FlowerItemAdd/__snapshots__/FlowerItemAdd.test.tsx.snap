// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FlowerItemAdd should render successfully 1`] = `
<div
  class="flower-item"
>
  <div
    class="card text-primary"
  >
    <div
      class="shelf-item-new"
    >
      <div
        class="card-title h5"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
        <span
          class="icon-prefix"
        >
          Add New Flower
        </span>
      </div>
    </div>
  </div>
</div>
`;
